### ISSUE TITLE
ClassCastException: java.lang.Integer cannot be cast to java.lang.String

### ISSUE THREAD

**Issue by seanf** [arturbosch/detekt#89]
Detekt version: M11. Might be related to #82. I'm happy to try again with M12 when it's out.

When this happens, it also causes the build to hang, so there seems to be a secondary problem with the exception handling for the CompletableFuture.

Workaround: change `indentSize: 4` to `indentSize: "4"` in detekt.yml, but it does seem like it should allow numbers.

```
[INFO] --- maven-antrun-plugin:1.8:run (detekt) @ zanata-model ---
[INFO] Executing tasks

detekt:
   [detekt] Exception in thread "main" java.util.concurrent.CompletionException: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
   [detekt] 	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
   [detekt] 	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
   [detekt] 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
   [detekt] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
   [detekt] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
   [detekt] 	at java.lang.Thread.run(Thread.java:748)
   [detekt] Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
   [detekt] 	at io.gitlab.arturbosch.detekt.formatting.Indentation$indent$1.invoke(Indentation.kt:31)
   [detekt] 	at io.gitlab.arturbosch.detekt.formatting.Indentation$indent$1.invoke(Indentation.kt:24)
   [detekt] 	at io.gitlab.arturbosch.detekt.api.Rule.withConfig(Rule.kt:68)
   [detekt] 	at io.gitlab.arturbosch.detekt.formatting.Indentation.<init>(Indentation.kt:31)
   [detekt] 	at io.gitlab.arturbosch.detekt.formatting.FormattingProvider.instance(FormattingProvider.kt:30)
   [detekt] 	at io.gitlab.arturbosch.detekt.api.RuleSetProvider$DefaultImpls.buildRuleset(RuleSetProvider.kt:28)
   [detekt] 	at io.gitlab.arturbosch.detekt.formatting.FormattingProvider.buildRuleset(FormattingProvider.kt:10)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.Detektor.analyze(Detektor.kt:48)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.Detektor.access$analyze(Detektor.kt:11)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:26)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.Detektor$run$1$$special$$inlined$map$lambda$1.invoke(Detektor.kt:11)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.TasksKt$runAsync$1.invoke(Tasks.kt:20)
   [detekt] 	at io.gitlab.arturbosch.detekt.core.TasksKt$task$1.get(Tasks.kt:24)
   [detekt] 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
   [detekt] 	... 3 more
```

My detekt.yml file:
```
autoCorrect: true

build:
  warningThreshold: 5
  failThreshold: 10
  weights:
    complexity: 2
    formatting: 0
    LongParameterList: 1
    comments: 0.5

potential-bugs:
  active: true
  DuplicateCaseInWhenExpression:
    active: true
  EqualsWithHashCodeExist:
    active: true
  ExplicitGarbageCollectionCall:
    active: true

exceptions:
  active: false

empty-blocks:
  active: true

complexity:
  active: true
  LongMethod:
    threshold: 20
  LongParameterList:
    threshold: 5
  LargeClass:
    threshold: 150
  ComplexMethod:
    threshold: 10
  TooManyFunctions:
    threshold: 10
  ComplexCondition:
    threshold: 3

code-smell:
  active: true
  FeatureEnvy:
    threshold: 0.5
    weight: 0.45
    base: 0.5

formatting:
  active: true
  useTabs: false
  Indentation:
    active: true
    indentSize: 4
  ConsecutiveBlankLines:
    active: true
    autoCorrect: true
  MultipleSpaces:
    active: true
    autoCorrect: true
  SpacingAfterComma:
    active: true
    autoCorrect: true
  SpacingAfterKeyword:
    active: true
    autoCorrect: true
  SpacingAroundColon:
    active: true
    autoCorrect: true
  SpacingAroundCurlyBraces:
    active: true
    autoCorrect: true
  SpacingAroundOperator:
    active: true
    autoCorrect: true
  TrailingSpaces:
    active: true
    autoCorrect: true
  UnusedImports:
    active: true
    autoCorrect: true
  OptionalSemicolon:
    active: true
    autoCorrect: true
  OptionalUnit:
    active: true
    autoCorrect: true
  ExpressionBodySyntax:
    active: false
    autoCorrect: true
  ExpressionBodySyntaxLineBreaks:
    active: false
    autoCorrect: true

style:
  active: true
  WildcardImport:
    active: true
  MaxLineLength:
    active: true
    maxLineLength: 120
  NamingConventionViolation:
    active: true
    variablePattern: '^(_)?[a-z$][a-zA-Z$0-9]*$'
    constantPattern: '^([A-Z_]*|serialVersionUID)$'
    methodPattern: '^[a-z$][a-zA-Z$0-9]*$'
    classPattern: '[A-Z$][a-zA-Z$]*'
    enumEntryPattern: '^[A-Z$][a-zA-Z_$]*$'

comments:
  active: true
  CommentOverPrivateMethod:
    active: true
  CommentOverPrivateProperty:
    active: true
  NoDocOverPublicClass:
    active: false
  NoDocOverPublicMethod:
    active: false

# *experimental feature*
# Migration rules can be defined in the same config file or a new one
migration:
  active: false
  imports:
    # your.package.Class: new.package.or.Class
    # for example:
    # io.gitlab.arturbosch.detekt.api.Rule: io.gitlab.arturbosch.detekt.rule.Rule
```

**Comment by arturbosch** [comment]
I think this must have been fixed in this commit 10872609d38d16d212e947834c59fbf4b235bb3b, will check again, thx.

**Comment by arturbosch** [comment]
I fixed. Added more test cases to ensure it will not break again!
